Bio/PDB/mmtf/mmtfio.py

```diff
@@ -98,12 +98,14 @@ def _save_structure(self, filepath, select):
 
         # The header information is missing for some structure objects
         header_dict = defaultdict(str, self.structure.header)
```

Comment:
I would simply keep this as a dict. The defaultdict will always return a value even if the key is not in the dictionary. That way you could simplify the code below like this:

```python
header_dict = dict(self.structure.header)  # copy
if "resolution" not in header_dict:
    header_dict["resolution"] = None

if "structure_method" not in header_dict:
    header_dict["structure_method"] = []

# Later around ln 123-ish
if "biomoltrans" in header_dict:
    # do the whole matrix stuff